presto-hive/src/test/java/com/facebook/presto/hive/benchmark/FileFormat.java

```diff
@@ -432,4 +450,43 @@ private static Properties createSchema(HiveStorageFormat format, List<String> co
                 .collect(joining(":")));
         return schema;
     }
+
+    private static class PrestoRcFileFormatWriter
+            implements FormatWriter
+    {
+        private final RcFileWriter writer;
+        private int[] positions = new int[0];
+
+        public PrestoRcFileFormatWriter(File targetFile, List<Type> types, RcFileEncoding encoding, HiveCompressionCodec compressionCodec)
+                throws IOException
+        {
+            writer = new RcFileWriter(
+                    new OutputStreamSliceOutput(new FileOutputStream(targetFile)),
+                    types,
+                    encoding,
+                    compressionCodec.getCodec().map(Class::getName),
+                    new AircompressorCodecFactory(new HadoopCodecFactory(getClass().getClassLoader())),
+                    ImmutableMap.of());
+        }
+
+        @Override
+        public void writePage(Page page)
+                throws IOException
+        {
+            if (page.getPositionCount() > positions.length) {
+                positions = new int[page.getPositionCount() * 2];
```

Comment:
`page.getPositionCount() * 2` could potentially exceed INT_MAX. Additionally, I feel `max(page.PositionCount(), positions.length * 2)` is a better choice.